Oqtane.Client/Modules/Controls/LocalizableComponent.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using Microsoft.AspNetCore.Components;
+using Microsoft.Extensions.Localization;
+using Oqtane.Shared;
+
+namespace Oqtane.Modules.Controls
+{
+    public class LocalizableComponent : ModuleControlBase
+    {
+        private IStringLocalizer _localizer;
+
+        [Parameter]
+        public string ResourceKey { get; set; }
+
+        protected bool IsLocalizable { get; private set; }
+
+        protected string Localize(string name)
+        {
+            if (!IsLocalizable)
+            {
+                return null;
+            }
+
+            var key = $"{ResourceKey}.{name}";
+ 
+            return _localizer?[key] ?? key;
+        }
+
+        protected override void OnParametersSet()
+        {
+            if (!String.IsNullOrEmpty(ResourceKey))
+            {
+                if (ModuleState?.ModuleType != null)
+                {
+                    var moduleType = Type.GetType(ModuleState.ModuleType);
+                    var localizerTypeName = $"Microsoft.Extensions.Localization.IStringLocalizer`1[[{moduleType.AssemblyQualifiedName}]], Microsoft.Extensions.Localization.Abstractions";
```

Comment:
Are you able to use https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.localization.istringlocalizerfactory.create?view=dotnet-plat-ext-3.1#Microsoft_Extensions_Localization_IStringLocalizerFactory_Create_System_Type_ instead?

Reply:
Oooh, how did I forgot that we already have `IStringLocalizerFactory.Create()` ;) I will try to simplify this soon

Thanks again

Reply:
I think the issue with `IStringLocalizerFactory.Create()` will create a new instance of the localizer at that point of time, but the aim to get the `IStringLocalizer` that already registered in the DI